content/ConnectingToDatabase/ConnectingToDatabase.q.md

```diff
@@ -0,0 +1,673 @@
+
+
+/// type=SS, answer=[4]
+
+What does the `catch` block do on line 14?
+
+- It checks the `$dsn` variable.
+
+- It evaluates the `if` statement.
+
+- It throws the value of the `$conn` object.
+
+- It displays the string `Unable to establish connection.` if the `if` statement evaluates to `true`.
+
+- It displays the string `Unable to establish a connection.` if there is a problem with the connection.
+
+
+/// type=MS, answer=[2,3]
+
+On line 3, remove the statement `$db = 'MyDatabase';`. Execute the program. What are the error messages?
+
+- Undefine variable: `db` on line 4
+
+- Undefined variable: `db` on line 9
+
+- Unable to establish a connection.
+
+- Undefined variable: `conn` on line 13
+
+- syntax error, unexpected `=`, expecting end of file on line 9
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=SS, answer=[4]
+
+Which statement best describes `PDO_PGSQL`?
+
+- It is a method that copies data from a file into the table.
+
+- It is a driver that enables access from PHP to MS SQL server.
+
+- It is a driver that enables access from PHP to MySQL databases.
+
+- It is a driver that enables access from PHP to PostgreSQL databases.
+
+- It is a method that copies data from the database table into PHP array.
+
+
+/// type=MS, answer=[1,2,3,4]
+
+Which of the following are elements of `PDO_PGSQL Data Source Name (DSN)`?
+
+- `host`
+
+- `port`
+
+- `user`
+
+- `pgsql:`
+
+- `databaseName`
+
+
+/// type=SS, answer=[5]
+
+Which `PDO_PGSQL DSN` element holds the name of the database?
+
+- `db`
+
+- `host`
+
+- `port`
+
+- `pgsql:`
+
+- `dbname`
+
+
+/// type=SS, answer=[2]
+
+Which `PDO_PGSQL DSN` element holds the name of the user?
+
+- `name`
+
+- `user`
+
+- `host`
+
+- `dbname`
+
+- `username`
+
+
+/// type=SS, answer=[5]
+
+Which statement best describes a `PDO` class?
+
+- It handles an exception if any error occurs.
+
+- It evaluates the connection in a PHP application.
+
+- It sets the parameters between PHP and a database server.
+
+- It allows unstable connection between PHP and a database.
+
+- It represents a connection between PHP and a database server.
+
+
++++
+
+
++++
+
+### Part 3: Finding and Fixing Errors
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+    $host='localhost';
+    $db = 'MyDatabase';
+    $port = '5432';
+    $username = 'postgres';
+    $password = 'Admin01';
+    
+    $dsn = "host=$host;port=$port;dbname=$db;user=$username;password=$password";
+
+    try{
+        $conn = new PDO($dsn);
+        if($conn){
+            echo "Successfully connected!"; 
+        }
+    }
+    catch(Exception $e){
+        echo "Unable to establish a connection.";
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- `An error occurred!`
+
+- No output is displayed.
+
+- `Successfully connected!`
+
+- `"Successfully connected!"`
+
+- `Unable to establish a connection.`
+
+
+/// type=MS, answer=[2,4]
+
+Which statements best describe the error?
```

Comment:
please replace `best` with something else... In question construction when we use the word `best` it means that we only require one answer which is considered as the best answer but in here we require two answers. Please check. Thanks.